dozer/cogs/tba.py

```diff
@@ -37,100 +41,271 @@ def __init__(self, bot):
+                            "https://www.chiefdelphi.com/media/photos/{foreign_key}",
+                            "https://www.chiefdelphi.com/media/img/{image_partial}"
+                        ),
+                        "imgur": (
+                            "Imgur",
+                            "https://imgur.com/{foreign_key}",
+                            "https://i.imgur.com/{foreign_key}.png"
+                        ),
+                        "instagram-image": (
+                            "instagram",
+                            "https://www.instagram.com/p/{foreign_key}",
+                            "https://www.instagram.com/p/{foreign_key}/media"
+                        ),
+                        "grabcad": (
+                            "GrabCAD",
+                            "https://grabcad.com/library/{foreign_key}",
+                            "{model_image}"
+                        )
+                    }.get(media.type, (None, None, None))
+                    media.details['foreign_key'] = media.foreign_key
+                    page = discord.Embed(title=base + name, url=url.format(**media.details))
+                    page.set_image(url=img_url.format(**media.details))
+                    pages.append(page)
+
+            if len(pages):
+                await paginate(ctx, pages)
+            else:
+                await ctx.send(f"No media for team {team_num} found in {year}!")
+
+        except aiotba.http.AioTBAError:
+            raise BadArgument("Couldn't find data for team {}".format(team_num))
+
+    media.example_usage = """
+    `{prefix}tba media 971 2016` - show available media from team 971 Spartan Robotics in 2016
+    """
+
+    @tba.command()
+    @bot_has_permissions(embed_links=True)
+    async def awards(self, ctx, team_num: int, year: int = None):
+        """Gets a list of awards the specified team has won during a year. """
+        with ctx.typing():
+            try:
+                awards_data = await self.session.team_awards(team_num, year=year)
+                events_data = await self.session.team_events(team_num, year=year)
+                event_key_map = {event.key: event for event in events_data}
+            except aiotba.http.AioTBAError:
+                raise BadArgument("Couldn't find data for team {}".format(team_num))
+
+            pages = []
+            for award_year, awards in itertools.groupby(awards_data, lambda a: a.year):
+                e = discord.Embed(title=f"Awards for FRC Team {team_num} in {award_year}:", color=discord.Color.blurple())
+                for event_key, event_awards in itertools.groupby(list(awards), lambda a: a.event_key):
+                    event = event_key_map[event_key]
+                    e.add_field(name=f"{event.name} [{event_key}]",
+                                value="\n".join(map(lambda a: a.name, event_awards)), inline=False)
+
+                pages.append(e)
+            if len(pages) > 1:
```

Comment:
This should be outside of the typing with block, so that Dozer stops typing after the message is sent instead of when the paginator stops.